File: java-samples/Advanced/snakesandladders-cordapp/oracle-service/src/main/java/net/corda/samples/snl/service/DiceRollService.java
package net.corda.samples.snl.service;

import net.corda.core.crypto.TransactionSignature;
import net.corda.core.node.AppServiceHub;
import net.corda.core.node.services.CordaService;
import net.corda.core.serialization.SingletonSerializeAsToken;
import net.corda.core.transactions.FilteredTransaction;
import net.corda.core.transactions.FilteredTransactionVerificationException;

import java.util.HashMap;
import java.util.Map;
import java.util.Random;


@CordaService
public class DiceRollService extends SingletonSerializeAsToken {

    private final AppServiceHub serviceHub;
    private Map<String, Integer> rollMap;

    public DiceRollService(AppServiceHub serviceHub) {
        this.serviceHub = serviceHub;
        this.rollMap = new HashMap<>();
    }

    public Integer diceRoll (String player){
        Random ran = new Random();
        int roll =  ran.nextInt(6) + 1;
        rollMap.put(player, roll);
        return roll;
    }

    public TransactionSignature sign(FilteredTransaction transaction) throws FilteredTransactionVerificationException {

        transaction.verify();

        boolean isValid = true; // Additional verification logic here.

        if(isValid){
            return serviceHub.createSignature(transaction, serviceHub.getMyInfo().getLegalIdentities().get(0).getOwningKey());
        }else{
            throw new IllegalArgumentException();
        }
    }
}

File: kotlin-samples/Advanced/snakesandladders-cordapp/oracle-service/src/main/kotlin/net/corda/samples/snl/diceservice/DiceRollService.kt
package net.corda.samples.snl.diceservice


import net.corda.core.crypto.TransactionSignature
import net.corda.core.node.AppServiceHub
import net.corda.core.node.services.CordaService
import net.corda.core.serialization.SingletonSerializeAsToken
import net.corda.core.transactions.FilteredTransaction
import net.corda.core.transactions.FilteredTransactionVerificationException
import java.util.HashMap
import java.util.Random


@CordaService
class DiceRollService(serviceHub: AppServiceHub) : SingletonSerializeAsToken() {
    private val serviceHub: AppServiceHub
    private val rollMap: MutableMap<String, Int>
    fun diceRoll(player: String): Int {
        val ran = Random()
        val roll = ran.nextInt(6) + 1
        rollMap[player] = roll
        return roll
    }

    @Throws(FilteredTransactionVerificationException::class)
    fun sign(transaction: FilteredTransaction): TransactionSignature {
        transaction.verify()
        val isValid = true // Additional verification logic here.
        return if (isValid) {
            serviceHub.createSignature(transaction, serviceHub.myInfo.legalIdentities.get(0).owningKey)
        } else {
            throw IllegalArgumentException()
        }
    }

    init {
        this.serviceHub = serviceHub
        rollMap = HashMap()
    }
}
